config/config.properties

```diff
@@ -427,24 +427,25 @@ hdfs.hdfsUrl=localhost:9000
 
 ##region Extended S3 settings
 
-# URL where the extended S3 cluster is accessible at.
+# URI to specify the complete extended S3 client configurations (excludes bucket and prefix), in the format of
+# <protocol>://<host>[:<port>][/path][?<param-key>=<param-value>][&<param-key>=<param-value>][...]
+# Refer https://github.com/EMCECS/ecs-object-client-java/wiki/Config-URI-format for more details.
 # This value must be the same for all Pravega SegmentStore instances in this cluster.
-extendeds3.url=localhost:9020
-
-# Prefix in extended S3 cluster where all Pravega-related data for this cluster is following.
-# This value must be the same for all Pravega SegmentStore instances in this cluster.
-# extendeds3.prefix=
-
-# ACCESS_KEY_ID to access the extended S3 cluster
-# extendeds3.accessKey=
-
-# SECRET_KEY to access the extended S3 cluster
-# extendeds3.secretKey=
+#
+# Supported parameters include but are not limited to:
+#   url: where the extended S3 cluster is accessible at, e.g. http://localhost:9020
+#   identity: the access key to access the extended S3 cluster, e.g. identity=user
+#   secretKey: the secret key to access the extended S3 cluster, e.g. secretKey=password
+extendeds3.configUri=http://localhost:9020?identity=user&secretKey=password
 
 # Shared extended S3 bucket where the data is stored.
 # This value must be the same for all Pravega SegmentStore instances in this cluster.
 # extendeds3.bucket=
 
+# Prefix in extended S3 cluster where all Pravega-related data for this cluster is following.
```

Comment:
Need to update the comment as it's a bit incorrect. A prefix in this case is a prefix that will be added to every object created by this instance of segment store in a configured S3 bucket. Using prefix allows sharing the same bucket between different Pravega clusters and also differentiate objects within a bucket created by Pravega from other objects. 

Also please mention that the prefix is optional and can be left empty.

Reply:
fixed